File: app/src/main/java/com/zerostress/manager/models/FriendRequest.java
package com.zerostress.manager.models;

public class FriendRequest {
    private String id;
    private String fromUserId;
    private String fromUserName;
    private String toUserId;
    private String status; // pending, accepted, rejected
    private long timestamp;

    public FriendRequest() {}

    public FriendRequest(String fromUserId, String fromUserName, String toUserId) {
        this.fromUserId = fromUserId;
        this.fromUserName = fromUserName;
        this.toUserId = toUserId;
        this.status = "pending";
        this.timestamp = System.currentTimeMillis();
    }

    public String getId() { return id; }
    public String getFromUserId() { return fromUserId; }
    public String getFromUserName() { return fromUserName; }
    public String getToUserId() { return toUserId; }
    public String getStatus() { return status; }
    public long getTimestamp() { return timestamp; }

    public void setId(String id) { this.id = id; }
    public void setStatus(String status) { this.status = status; }
}

File: app/src/main/java/com/zerostress/manager/models/FriendRequest.kt
package com.zerostress.manager.models

import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class FriendRequest(
    var id: String = "",
    var fromUserId: String = "",
    var fromUserName: String = "",
    var toUserId: String = "",
    var status: String = "pending",
    var timestamp: Long = 0L
) {
    companion object {
        fun create(fromUserId: String, fromUserName: String, toUserId: String): FriendRequest =
            FriendRequest(
                fromUserId = fromUserId,
                fromUserName = fromUserName,
                toUserId = toUserId,
                status = "pending",
                timestamp = System.currentTimeMillis()
            )
    }
}
